### ISSUE TITLE
Rename Direct Access Mode in the data source settings

### ISSUE THREAD

**Issue by daniellee** [grafana/grafana#11391]
Currently, the two choices for Access Mode in the data source settings are Proxy and Direct. A lot of users choose Direct and then create issues or questions on the community site when they fail to connect to their data source. Direct sounds better than Proxy if you don't know what the two terms mean in the Grafana context.

We could significantly decrease the support burden by renaming Direct to something else or by doing a UX change that explains why Proxy should be chosen > 90% of the time.

**Comment by marefr** [comment]
From the docs "Proxy = access via Grafana backend, Direct = access directly from browser."

**Comment by DanCech** [comment]
@marefr I think you mean a warning when browser/direct mode is selected

I like the sound of browser vs server mode. A popover in the ui to explain them sounds like a good idea too

**Comment by marefr** [comment]
@DanCech yes that's what I meant :+1:

**Comment by torkelo** [comment]
Is there not already a popover that contains that docs info?

browser/server might be slightly more intuitive. But tricky to change now and make it backward compatible. So maybe adding  better error handling and tooltips / tips is more time effective.

**Comment by marefr** [comment]
There's a popover visible on hover of info icon, but I wonder how many users actually looking at that.

![image](https://user-images.githubusercontent.com/1668778/38073906-41627e8e-332d-11e8-8b7a-f49196b47ca2.png).

Will take a quick look into possibility to change text but preserving value in the select box. I still agree that better error handling and tooltips are needed.

**Comment by marefr** [comment]
Read up on CORS and found the [following information](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS):

>In particular, a request is preflighted if any of the following conditions is true:
> 
>* If the request uses any of the following methods:
>   * PUT
>   * DELETE
>   * CONNECT
>   * OPTIONS
>   * TRACE
>   * PATCH
> * Or if, apart from the headers set automatically by the user agent (for example, Connection, User-Agent, or any of the other header with a name defined in the Fetch spec as a “forbidden header name”), the request includes any headers other than those which the Fetch spec defines as being a “CORS-safelisted request-header”, which are the following:
>   * Accept
>   * Accept-Language
>   * Content-Language
>   * Content-Type (but note the additional requirements below)
>   * Last-Event-ID
>   * DPR
>   * Save-Data
>   * Viewport-Width
>   * Width
> * Or if the Content-Type header has a value other than the following:
>   * application/x-www-form-urlencoded
>   * multipart/form-data
>   * text/plain

I think that one of the biggest concerns regarding CORS with our implementation of *test datasource* is that they don't trigger preflighted requests, rather simple requests as described in referenced link.

**Comment by marefr** [comment]
Have been experimenting with something like this when we receive a XmlHttpRequest error in edit data source. When `error.statusText` and `error.message` is empty and the data source in question have `access=direct` configured we can show an error similar to this:

![image](https://user-images.githubusercontent.com/1668778/38321038-a60816ba-3836-11e8-9c55-12c4be4a3562.png)

Thoughts @grafana/grafana-labs-core-devs ?

**Comment by marefr** [comment]
Alternatives for showing more information based on what access method has been selected. This is the preferred since it can be used for all data sources including plugins.

![image](https://user-images.githubusercontent.com/1668778/38324503-d0f6b70a-3840-11e8-826d-23f9e45d17b3.png)

![image](https://user-images.githubusercontent.com/1668778/38324509-d5dd173c-3840-11e8-9f1b-cb5503e2704d.png)

The info box is not optimal for displaying a lot of text so some UX input would be nice here.

**Comment by torkelo** [comment]
How is the info box triggered (when is it show)? Way to big to always show it.

Maybe we ca use the same drop down we use for permissions that have a description as well ?  Might be tricky thought as that is in react

**Comment by marefr** [comment]
The blue info boxes are always shown in this example. The red error box above was an attempt to catch a possible CORS issue, but requires each data source to implement additional logic then testing data source. 

Had a discussion with team about this right now and we've decided to try and start with better naming. The react dropdown could be interesting, but maybe tricky as you said. Will do a quick POC of using that drop down.

**Comment by xlson** [comment]
Couldn't we rename the fields while still supporting the old values as well? Perhaps using `default` and `browser` instead so that it's obvious what you probably should be using. I feel much of the confusion now comes from proxy being something non-default while direct sounds more assertive and default-y.

Actually, couldn't we remove the settings option completely and have it be something that's only available if you enable it through server settings?

**Comment by marefr** [comment]
Managed to use a different label while still having the old values to be backward compatible. This makes it a lot more understandable! 

![image](https://user-images.githubusercontent.com/1668778/38366607-9a7a8986-38e0-11e8-9b53-21bcc118fbc4.png)

![image](https://user-images.githubusercontent.com/1668778/38366616-a0a65f7e-38e0-11e8-9552-3ffce7262fd8.png)

Using the react dropdown with description is too tricky for me. I tried a different approach in comparison to showing a tooltip when hover over the info icon and instead letting the user click a button to expand/collapse the help section:

![image](https://user-images.githubusercontent.com/1668778/38366668-d489b49e-38e0-11e8-9cb1-6db93a6a97a5.png)

![image](https://user-images.githubusercontent.com/1668778/38366673-d96fc6f6-38e0-11e8-8e80-772e8e315615.png)

I kind of like this approach since the help button stands out a little bit more than the previous info icon and also allows more content than in a tooltip.

Thoughts @grafana/grafana-labs-core-devs?

@xlson regarding hiding stuff under an advanced section is probably a good idea, but think that we can create a separate issue for that since we'll probably need UX etc.
